### PR TITLE
send code to quickbench

### LINKED ISSUE #581: quick-bench.com integration
As discussed with @FredTingaud and now we have Google Benchmark available we should have bidirectional links to quick-bench.com :)

### PR DESCRIPTION
**Pull request by partouf**
Fixes #581 

Has some issues:
* Text is not vertically aligned with other buttons.
* Button image comes from Github, couldn't find how to link to pngs?
* Not entirely sure about the Unicode encoding, but it seems to be working.

Corresponding Quick-bench PR: https://github.com/FredTingaud/quick-bench-front-end/pull/29

### CODE REVIEW

**Comment by apmorton** [comment]
Did some digging and the quickbench logo comes from here:

https://www.freepik.com/free-vector/chronometer-timer-collection_717989.htm

It is available free with attribution, which has been done on the quickbench site - maybe we should do the same somewhere here?

In addition, it is available as a vector drawing - which is preferable to linking to a png off GitHub.

I can produce a nicely compact svg and post it here later today.

**Comment by partouf** [comment]
> Did some digging and the quickbench logo comes from here:
> 
> https://www.freepik.com/free-vector/chronometer-timer-collection_717989.htm
> 
> It is available free with attribution, which has been done on the quickbench site - maybe we should do the same somewhere here?
> 
> In addition, it is available as a vector drawing - which is preferable to linking to a png off GitHub.
> 
> I can produce a nicely compact svg and post it here later today.

Ahh, excellent! Thanks for the digging. Svg would be much better yes

**Comment by apmorton** [comment]
optimized quickbench.svg here: https://gist.github.com/apmorton/21aeb90471e731c6f53ba21cde035005

**Comment by FredTingaud** [comment]
Thanks for the PR, it is online on quick-bench side.

**Comment by apmorton** [comment]
I propose we change the unicode escaping to be done in the json itself.

```javascript
var quickBenchState = {
    text: this.getSource()
};

var json = JSON.stringify(quickBenchState);
json = json.replace(/[\u007F-\uFFFF]/g, function(chr) {
    return "\\u" + ("0000" + chr.charCodeAt(0).toString(16)).substr(-4);
});

var link = 'http://quick-bench.com/#' + btoa(json);
```

On the quickbench side, the code simply becomes:
```javascript
return JSON.parse(atob(base64ascii));
```
`JSON.parse` handles unescaping the unicode characters.

The end result is the same, but it uses a completely standard escaping mechanism that has no potential for interfering with user source code.

For example, the following would be mangled by the current unicode escaping on the quickbench side:

```C++
auto str = "&$abcdef";
```

File: static/panes/editor.js [hunk 1]
```diff
@@ -475,13 +481,28 @@ Editor.prototype.b64UTFEncode = function (str) {
     }));
 };
 
+Editor.prototype.b64EncodeWithHtmlUnicode = function (str) {
+    return btoa(str.replace(/[\u00A0-\u2666]/g, function (c) {
```
Comment (mattgodbolt) [inline static/panes/editor.js]: Neat way to do this!

**Comment by mattgodbolt** [comment]
@apmorton 's suggestions work too, I'm delighted to get this feature in howsoever it's achieved. Thanks @FredTingaud too :)

**Comment by apmorton** [comment]
Here is a code example showing the unicode escaping conflict that could (but probably never actually would) happen in user code prior to the changes I made:
https://godbolt.org/z/o8XSHK

@FredTingaud the unicode changes do require action on your end to actually prevent the conflict from happening.

https://github.com/FredTingaud/quick-bench-front-end/blob/master/src/Benchmark.js#L225 can simply be changed to
```javascript
return JSON.parse(atob(base64ascii));
```

**Comment by FredTingaud** [comment]
I'll update it and put the fix online in the next 4h :)

**Comment by FredTingaud** [comment]
Done on my side.
Although I didn't test anything!

**Comment by partouf** [comment]
Looks to be working good from staging environment, will check when we can launch this live.

**Pull request merged**